data/modules/Taxi/Taxi.lua

```diff
@@ -254,7 +255,12 @@ local makeAdvert = function (station)
 	if #nearbysystems == 0 then return end
 	location = nearbysystems[Engine.rand:Integer(1,#nearbysystems)]
 	local dist = location:DistanceTo(Game.system)
-	reward = ((dist / max_taxi_dist) * typical_reward * (group / 2) * (1+risk) * (1+3*urgency) * Engine.rand:Number(0.8,1.2))
+
+        local cargo = Game.player.totalCargo * 0.81 --the rest 0.19 is used for fuel.
+	local rewardRatio = 2/3 -- For getting that ratio of current average trading profit
+        local profit = math.max(Calibration.profit(cargo) * rewardRatio, Calibration.minProfit * rewardRatio)
+
+	reward = ((dist / max_taxi_dist) * typical_reward * (group / 2) * (1+risk) * (1+3*urgency) * Engine.rand:Number(0.8,1.2)) * profit/5666 --5666 is the current estimated profit from taxi module.
```

Comment:
I'm not certain about the math here; in this case I think I'd rather calculate some mapping between the number of people to be taxied and an equivalent "required cargo space" and use that to set an equivalent profit; e.g. 2 passengers = 6 cargo space, post processed = 5,200 reward or something like that. Again, scaling mission reward by player's current cargo space is *independent* of any of the mission's actual parameters and just feels cheap that you have to buy a bigger ship to get more money out of the same exact mission.

Reply:
Assassination.lua is that plus the missiles profit. 'profit' is either 2/3 of current player's ship average profit, calculated by its free cargo, or 2/3 of the minimum trading profit, that is the trading profit as if ship could trade full offer of major exported precious metals. Whatever is maximum between these two.

Else, my reply is the same of that i have in assassinations code.

Having said that, missions rewards have nothing to do with exponential and the main scope of this PR. Modify or reject the code at will.